src/node_options.h

```diff
@@ -0,0 +1,57 @@
+#ifndef SRC_NODE_OPTIONS_H_
+#define SRC_NODE_OPTIONS_H_
+
+#include "node_version.h"
+#include "util.h"
+#include "getopt.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+namespace node {
+
+class NodeOptions {
+ public:
+  static bool print_eval;
+  static bool force_repl;
+  static bool trace_deprecation;
+  static bool throw_deprecation;
+  static bool abort_on_uncaught_exception;
+  static bool trace_sync_io;
+  static const char* eval_string;
+  static unsigned int preload_module_count;
+  static const char** preload_modules;
+  static bool use_debug_agent;
+  static bool debug_wait_connect;
+  static int debug_port;
+  static bool no_deprecation;
+#if defined(NODE_HAVE_I18N_SUPPORT)
+  // Path to ICU data (for i18n / Intl)
+  static const char* icu_data_dir;
+#endif
```

Comment:
It might be nicer to turn these into fields and then have a global `node_options` or `cli_options` variable.  Either that or make it impossible to instantiate by adding a `NodeOptions() = delete`.